### ISSUE TITLE
Work Item Print Not Working All Of A Sudden

### ISSUE THREAD

**Issue by johnkrause05** [MrTarantula/WIPrint#21]
Just yesterday, I go to print using Google Chrome and my work item print is blank.  I go to the configuration settings and it has fields selected.  I try to add and remove some fields and then save, still shows blank.  I tried on microsoft edge and it works correctly.  It seems to have broke on google chrome within the last few days?  This always worked before, this is something I use very often.

Thanks

![work item blank](https://user-images.githubusercontent.com/25367835/37523700-1e79d496-28fe-11e8-8989-f1f546993fdc.jpg)
![work item print settings](https://user-images.githubusercontent.com/25367835/37523710-229cb0ca-28fe-11e8-9ea8-778f44e5e550.jpg)

**Comment by MrTarantula** [comment]
Can you open developer tools (F12) and try to print something. Do you receive an error?

**Comment by johnkrause05** [comment]
Thank you for your quick response.  Running the developer console and printing something gives no message at all.  I would be willing to share my screen in team viewer if that would be helpful.

John Krause | 810-229-9334 | john.krause@promessinc.com<mailto:john.krause@promessinc.com>

Promess Incorporated | 11429 Grand River Rd., Brighton, MI 48116

[Promess Logo]

 www.promessinc.com<http://www.promessinc.com>

 [Promess Twitter] <https://twitter.com/PromessInc>   [Promess YouTube] <http://www.youtube.com/user/PromessInc?feature=watch>   [Promess LinkedIn] <http://www.linkedin.com/company/promess>

**Comment by MrTarantula** [comment]
No, I don't have the time to devote to issues like this. I have a day job and this is just a side project. Have you tried clearing your cache in Chrome?

**Comment by johnkrause05** [comment]
O absolutely, my apologies 😊

I will try that, thanks a bunch.  Excellent job on the extension by the way, you did a great job.

John Krause | 810-229-9334 | john.krause@promessinc.com<mailto:john.krause@promessinc.com>

Promess Incorporated | 11429 Grand River Rd., Brighton, MI 48116

[Promess Logo]

 www.promessinc.com<http://www.promessinc.com>

 [Promess Twitter] <https://twitter.com/PromessInc>   [Promess YouTube] <http://www.youtube.com/user/PromessInc?feature=watch>   [Promess LinkedIn] <http://www.linkedin.com/company/promess>

**Comment by MrTarantula** [comment]
It's possible that it's taking too much time for VSTS to load the data before Chrome begins printing, especially if you're printing a lot of items or if your network or VSTS are slow. It has been an issue before, and I hacked a fix in. I don't have a good fix for it yet.

**Comment by johnkrause05** [comment]
I deleted my cache in chrome and still doesn't work.  When trying to use in firefox or edge, this now happens when I try to go to the settings page.  Any thoughts on this?

![taking longer than expected](https://user-images.githubusercontent.com/25367835/37542047-2c606a2c-2933-11e8-868b-51c9a435bf99.jpg)

**Comment by MrTarantula** [comment]
I'm not experiencing that problem in VSTS. Have you made any changes to your work item templates or processes?

![image](https://user-images.githubusercontent.com/11617455/37542319-c87f090c-292b-11e8-9dcd-80fdf40822ae.png)

**Comment by johnkrause05** [comment]
In google chrome I am able to load the settings page, but when trying to print it just prints a blank page.  When using edge (now doesn't work, worked previously) or firefox this happens.  Below is the console printout if this helps.

![console](https://user-images.githubusercontent.com/25367835/37542590-0f1331a0-2935-11e8-87da-c18032258f80.jpg)

**Comment by MrTarantula** [comment]
If you go to the network tab, then refresh, are there any files with 404 error?

**Comment by johnkrause05** [comment]
This is the results in edge.

![work item print network console](https://user-images.githubusercontent.com/25367835/37543854-5b72f20c-2939-11e8-8440-506de7c9a7e2.jpg)

**Comment by MrTarantula** [comment]
It looks like VSTS is giving you a 302 redirect when trying to load my extension. I'm not sure why that would happen. It could be VSTS, or (less likely) corporate firewall or antivirus. Do you have issues loading any other extensions?

**Comment by MKGNZ** [comment]
Same here this is what I get:

![image](https://user-images.githubusercontent.com/37227480/37663522-c286ea86-2c2f-11e8-9728-a053ac266f45.png)

**Comment by JamesHurburgh** [comment]
I am getting the same error as @MKGNZ, but only on work items with comments.  Printing work items without comments seems to be working fine.

**Comment by JamesHurburgh** [comment]
Looks like you need to check comment.revisedBy.name for nulls [here](https://github.com/MrTarantula/WIPrint/blob/0a47a38cd4e97047ed68254dc55871e352759b13/src/wiprint.ts#L251) and [here](https://github.com/MrTarantula/WIPrint/blob/0a47a38cd4e97047ed68254dc55871e352759b13/src/wiprint.ts#L275).

**Comment by MrTarantula** [comment]
`revisedBy.name` should never be null. If there is a comment, then there is an author. What's more likely is that `revisedBy.name` has been replaced with a new field. I'll have to dig through the ever-changing documentation and find it.

**Comment by JamesHurburgh** [comment]
Debugging in the Chrome console gives me an object with this structure:

revisedBy:
{
	displayName:""
	id:""
	imageUrl:""
	uniqueName:""
	url:""
}

**Comment by MrTarantula** [comment]
It also appears that instead of an array of `WorkItemComment` VSTS now gives me a `WorkItemComments` object with an array of `comment` properties. I think I have it sorted out in the latest release, which I have in the marketplace but not Github.

**Comment by JamesHurburgh** [comment]
Thanks very much for the fix.  That seems to have resolved the issue I was having.
